src/libs/kata-types/src/cpu.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::convert::TryFrom;
+use std::str::FromStr;
+
+use oci::LinuxCpu;
+
+/// A set of CPU ids.
+pub type CpuSet = crate::utils::u32_set::U32Set;
+
+/// A set of NUMA memory nodes.
+pub type NumaNodeSet = crate::utils::u32_set::U32Set;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Invalid CPU list: {0}")]
+    InvalidCpuSet(crate::Error),
+    #[error("Invalid NUMA memory node list: {0}")]
+    InvalidNodeSet(crate::Error),
+}
+
+/// Assigned CPU resources for a Linux container.
+#[derive(Default, Debug)]
+pub struct LinuxContainerCpuResources {
+    shares: u64,
+    period: u64,
+    quota: i64,
+    milli_sec: Option<u64>,
```

Comment:
Would `some_name_ms` or similar be clearer here?

Reply:
renamed to "calculated_vcpu_time_ms".